base/applications/dxdiag/system.c

```diff
@@ -225,32 +227,73 @@ InitializeSystemPage(HWND hwndDlg)
     if (GetLocaleInfo(LOCALE_SYSTEM_DEFAULT,LOCALE_SLANGUAGE , szTime, sizeof(szTime) / sizeof(WCHAR)))
         SendDlgItemMessageW(hwndDlg, IDC_STATIC_LANG, WM_SETTEXT, 0, (LPARAM)szTime);
 
+    /* prepare SMBIOS data */
+    SMBiosBuf = LoadSMBiosData(DmiStrings);
+
     /* set system manufacturer */
     szTime[0] = L'\0';
-    if (GetRegValue(HKEY_LOCAL_MACHINE, L"Hardware\\Description\\System\\BIOS", L"SystemManufacturer", REG_SZ, szTime, sizeof(szTime)))
+    Length = 0;
+    if (SMBiosBuf)
+    {
+        Length = GetSMBiosStringW(DmiStrings[SYS_VENDOR], szTime, sizeof(szTime) / sizeof(WCHAR), FALSE);
+    }
+    if (Length == 0)
+    {
+        Length = GetRegValue(HKEY_LOCAL_MACHINE, L"Hardware\\Description\\System\\BIOS", L"SystemManufacturer", REG_SZ, szTime, sizeof(szTime));
+    }
+    if (Length > 0)
     {
         szTime[199] = L'\0';
```

Comment:
199?

Reply:
It refers to the last character in the string, but that's not my code :smile:

Reply:
Yeah, in cases like that I would opt for `_countof() -1`

Reply:
Ok, since I'm at it, fixed in a separate commit.